lua/entities/base_wire_entity.lua

```diff
@@ -373,15 +376,14 @@ net.Receive( "wire_overlay_request", function( len, ply )
 end )
 
 hook.Add("PlayerDisconnected","wire_playersRequestingOverlay_cleanup",function(ply)
-	table.RemoveByValue(self.playersRequestingOverlayNumeric,ply)
+	table.RemoveByValue(ENT.playersRequestingOverlayNumeric,ply)
```

Comment:
This won't work. ENT 'should' be a nil global at this time of execution. I prefer my idea. Also I don't think PlayerDisconnected is reliable in all cases.

Reply:
ive never seen that hook fail to fire.

Reply:
Maybe I'm wrong there, but this still won't work.

Reply:
Yeah, at runtime ENT is set to nil. This will cause lua errors.

Reply:
ah, i will have to loop over all wire entities and remove the player out of the table if they are in it, when a player leaves.  I think this is the best place to put this expensive function, in a rarely called hook.

Reply:
You can either use entity discovery to fetch all entities, or you can add them to a list in the initialize hook and remove them from the list in the onremove hook. The former is easier, the latter is more efficient.